src/include/buffer.h

```diff
@@ -304,7 +310,6 @@ namespace buffer CEPH_BUFFER_API {
 
       void advance(int o);
       void seek(unsigned o);
-      bool operator!=(const iterator_impl& rhs) const;
```

Comment:
but @jdurgin i did break the ABI compatibility in #7688 which violates https://techbase.kde.org/Policies/Binary_Compatibility_Examples#Unexport_or_remove_a_class: the iterator_impl was referenced in the header file by the public APIs so, as a sequence it is exported as weak symbols. and in #7688, i fixed it by 

1. moving the implementation of public interface into the .cc file as you suggested and
1. hiding the template symbols by using `__attribute__ ((visibility ("hidden")))`

this helps to bring the backward compatibility of librados clients but breaks the forward compatibility of librados clients compiled with v9.1.0 and v9.2.0. that's why @smithfarm suggest we bump the so version.